File: go/300.500/405.ConvertANumberToHexadecimal.go
package main

import (
	"fmt"
	"strings"
)

func toHex(num int) string {
	if num == 0{
		return "0"
	}
	sb := strings.Builder{}
	for i:= 7;i>=0;i--{
		val := (num >> (4*i))&0xf
		if sb.Len() >0 || val > 0{
			if val < 10{
				sb.WriteByte('0'+ byte(val));
			}else{
				sb.WriteByte('a'+ byte(val-10));
			}
		}
	}
	return sb.String()
}

func main() {
	fmt.Println(toHex(26), "1a")
	fmt.Println(toHex(-1), "ffffffff")
}

File: 300.500/405.ConvertANumberToHexadecimal.cpp
#include <iostream>
#include <string>
using namespace std;

class ConvertANumberToHexadecimal{
public:
    string toHex(int num) {
        if (num == 0)
            return "0";
        string ret = "";
        for (int i = 7; i >= 0;--i) {
            int val = (num >> (4 * i)) & 0xf;
            if (ret != "" || val > 0) {
                ret.push_back(val < 10 ? ('0' + val) : ('a' + val - 10));
            }
        }
        return ret;
    }
};

int main(int argc, char const* argv[]) {
    ConvertANumberToHexadecimal cls;
    cout << cls.toHex(26) << " -> 1a" << endl;
    cout << cls.toHex(-1) << " -> ffffffff" << endl;
    return 0;
}
